### PR TITLE
Switch metrics from tgo to go-metrics

### PR DESCRIPTION
**Pull request by arnecls**
## The purpose of this pull request

Exchange tgo.metrics with go-metrics to improve metrics and allow different metrics provider.
Request as by issue #221

## Checklist

- [x] `make test` executed successfully
- [ ] unit test provided
- [ ] docs updated

### CODE REVIEW

File: core/metrics.go [hunk 1]
```diff
@@ -15,199 +15,147 @@
...
+	metricBuild.Update(int64(build))
+	go metrics.CaptureRuntimeMemStats(MetricsRegistry, time.Second)
 }
 
-// CountRouters increases the stream counter by 1
-func CountRouters() {
-	tgo.Metric.Inc(metricRouters)
+// NewSubRegistry creates a new, prefixed metrics registry that can be used
+// to register custom plugin metrics.
+func NewSubRegistry(prefix string) metrics.Registry {
```
Comment (ppar) [inline core/metrics.go]: Could NewSubRegistry() and NewPluginRegistry() be renamed to something like NewMetricsSubRegistry() and NewMetricsPluginRegistry()? Currently when you encounter references to these elsewhere in the code, nothing tells you that they're metrics-specific functions (other than the return value's type, which is invisible) - they look like central parts of Gollum's core instead.

**Comment by ppar** [comment]
Looks OK to me - however, I haven't worked closely with a lot of the code involved here.

File: core/metrics.go (hunk 1 above)
Comment (arnecls) [reply core/metrics.go]: good point

Comment (arnecls) [reply core/metrics.go]: adjusted. It's now `NewMetricsRegistry` and `NewMetricsRegistryForPlugin`.

**Comment by arnecls** [comment]
merged in order to move forward with 0.6

**Pull request merged**